### PR TITLE
Logging in cwlogs format for python sources

### PR DESCRIPTION
**Pull request by andreiko**
before:
```
WARNING:retry.api:

  Message: Call was not successful
  Method:  GET
  Path:    https://graph.facebook.com/v2.10/6054997905162/insights
  Params:  {'date_preset': 'lifetime', 'action_attribution_windows': 'default', 'fields': 'frequency,impressions,ad_id,social_impressions,unique_clicks,unique_social_clicks,inline_post_engagement,clicks,date_stop,spend,call_to_action_clicks,date_start,social_spend,social_clicks,reach,actions', 'time_increment': 1, 'action_breakdowns': 'action_type,action_link_click_destination'}

  Status:  400
  Response:
    {
      "error": {
        "message": "Unsupported get request.",
        "fbtrace_id": "ADRHdh4QifZ",
        "code": 100,
        "type": "GraphMethodException"
      }
    }
, retrying in 5 seconds...
WARNING:requests.packages.urllib3.connectionpool:Connection pool is full, discarding connection: graph.facebook.com
```

after:
```
{"time": "2017-11-22T19:33:30.629684Z", "level": "warning", "message": "Connection pool is full, discarding connection: graph.facebook.com", "data": {"logger": "requests.packages.urllib3.connectionpool", "program": "facebook-ads", "version": "0.12.9"}}
{"time": "2017-11-22T19:33:32.158741Z", "level": "warning", "message": "\n\n  Message: Call was not successful\n  Method:  GET\n  Path:    https://graph.facebook.com/v2.10/6054997905162/insights\n  Params:  {'fields': 'date_stop,spend,impressions,date_start,inline_post_engagement,social_impressions,reach,clicks,unique_clicks,call_to_action_clicks,social_clicks,ad_id,social_spend,frequency,unique_social_clicks,actions', 'date_preset': 'lifetime', 'action_attribution_windows': 'default', 'time_increment': 1, 'action_breakdowns': 'action_type,action_link_click_destination'}\n\n  Status:  400\n  Response:\n    {\n      \"error\": {\n        \"type\": \"GraphMethodException\",\n        \"fbtrace_id\": \"Fw15+eCuZpk\",\n        \"message\": \"Unsupported get request.\",\n        \"code\": 100\n      }\n    }\n, retrying in 5 seconds...", "data": {"logger": "retry.api", "program": "facebook-ads", "version": "0.12.9"}}
```

### CODE REVIEW

**Comment by cdemwell** [comment]
This is great work - Dean and I looked at it, as well as the facebook-ads changes. We discovered some important things to note in case of any investigation:

1. Of the three Python Sources, facebook-ads and marketo use python-source-resource. Google adwords does not - it implements its own resource running framework. It will therefore not automatically pick up anything from here.

2. There is some risk with merging a change like this during a code freeze while the original author is away. We think we can mitigate this with a feature branch at the facebook-ads level. We have no intention of redeploying Marketo, so it can run for now with the older python-source-resource library. As mentioned, adwords will not be affected.

3. We currently have abstractions in python-source-resource and also in python-source. Logging is not strictly speaking a resource issue, so it seems like we should consider how we structure our abstractions at some point. Dean and I think that it would be a good option to consider refactoring our Python libraries at some point, either choosing to separate them all into separate repos or merging them together (probably that one as it's easier to manage).

In conclusion, we are going to merge and publish this change.
